File: app/src/main/java/jp/hazuki/yuzubrowser/adblock/AdBlockActivity.java
/*
 * Copyright (C) 2017 Hazuki
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package jp.hazuki.yuzubrowser.adblock;

import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;

import java.util.List;

import jp.hazuki.yuzubrowser.R;
import jp.hazuki.yuzubrowser.adblock.fragment.AdBlockFragment;
import jp.hazuki.yuzubrowser.adblock.fragment.AdBlockImportFragment;
import jp.hazuki.yuzubrowser.adblock.fragment.AdBlockMainFragment;
import jp.hazuki.yuzubrowser.utils.app.ThemeActivity;

public class AdBlockActivity extends ThemeActivity implements AdBlockMainFragment.OnAdBlockMainListener, AdBlockFragment.AdBlockFragmentListener, AdBlockImportFragment.OnImportListener {
    public static final String ACTION_OPEN_BLACK = "jp.hazuki.yuzubrowser.adblock.AdBlockActivity.action.open.black";
    public static final String ACTION_OPEN_WHITE = "jp.hazuki.yuzubrowser.adblock.AdBlockActivity.action.open.white";
    public static final String ACTION_OPEN_WHITE_PAGE = "jp.hazuki.yuzubrowser.adblock.AdBlockActivity.action.open.whitepage";

    private static final String TAG_LIST = "list";


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.fragment_base);

        if (savedInstanceState == null)
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.container, new AdBlockMainFragment())
                    .commit();

        if (getIntent() != null && getIntent().getAction() != null) {
            switch (getIntent().getAction()) {
                case ACTION_OPEN_BLACK:
                    openBlackList();
                    break;
                case ACTION_OPEN_WHITE:
                    openWhiteList();
                    break;
                case ACTION_OPEN_WHITE_PAGE:
                    openWhitePageList();
                    break;
            }
        }
    }

    @Override
    public void openBlackList() {
        openList(AdBlockManager.TYPE_BLACK_TABLE);
    }

    @Override
    public void openWhiteList() {
        openList(AdBlockManager.TYPE_WHITE_TABLE);
    }

    @Override
    public void openWhitePageList() {
        openList(AdBlockManager.TYPE_WHITE_PAGE_TABLE);
    }

    private void openList(int type) {
        getSupportFragmentManager().beginTransaction()
                .replace(R.id.container, AdBlockFragment.newInstance(type), TAG_LIST)
                .addToBackStack("type:" + type)
                .commit();
    }

    @Override
    public void onImport(List<AdBlock> adBlocks) {
        Fragment fragment = getSupportFragmentManager().findFragmentByTag(TAG_LIST);
        if (fragment instanceof AdBlockFragment) {
            ((AdBlockFragment) fragment).addAll(adBlocks);
        }
    }


    @Override
    public void setFragmentTitle(int type) {
        switch (type) {
            case AdBlockManager.TYPE_BLACK_TABLE:
                setTitle(R.string.pref_ad_block_black);
                break;
            case AdBlockManager.TYPE_WHITE_TABLE:
                setTitle(R.string.pref_ad_block_white);
                break;
            case AdBlockManager.TYPE_WHITE_PAGE_TABLE:
                setTitle(R.string.pref_ad_block_white_page);
                break;
        }
    }

    @Override
    public void requestImport(Uri uri) {
        getSupportFragmentManager().beginTransaction()
                .replace(R.id.container, AdBlockImportFragment.newInstance(uri))
                .addToBackStack("")
                .commit();
    }

    @Override
    public String getExportFileName(int type) {
        switch (type) {
            case AdBlockManager.TYPE_BLACK_TABLE:
                return "black_list.txt";
            case AdBlockManager.TYPE_WHITE_TABLE:
                return "white_list.txt";
            case AdBlockManager.TYPE_WHITE_PAGE_TABLE:
                return "white_page_list.txt";
            default:
                throw new IllegalArgumentException();
        }
    }
}

File: module/adblock/src/main/java/jp/hazuki/yuzubrowser/adblock/ui/original/AdBlockActivity.kt
/*
 * Copyright (C) 2017-2019 Hazuki
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package jp.hazuki.yuzubrowser.adblock.ui.original

import android.net.Uri
import android.os.Bundle
import jp.hazuki.yuzubrowser.adblock.R
import jp.hazuki.yuzubrowser.adblock.repository.original.AdBlock
import jp.hazuki.yuzubrowser.adblock.repository.original.AdBlockManager
import jp.hazuki.yuzubrowser.ui.app.ThemeActivity

class AdBlockActivity : ThemeActivity(), AdBlockMainFragment.OnAdBlockMainListener, AdBlockFragment.AdBlockFragmentListener, AdBlockImportFragment.OnImportListener {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.fragment_base)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        if (savedInstanceState == null)
            supportFragmentManager.beginTransaction()
                    .replace(R.id.container, AdBlockMainFragment())
                    .commit()

        if (intent != null && intent.action != null) {
            when (intent.action) {
                ACTION_OPEN_BLACK -> openBlackList()
                ACTION_OPEN_WHITE -> openWhiteList()
                ACTION_OPEN_WHITE_PAGE -> openWhitePageList()
            }
        }
    }

    override fun openBlackList() {
        openList(AdBlockManager.TYPE_BLACK_TABLE)
    }

    override fun openWhiteList() {
        openList(AdBlockManager.TYPE_WHITE_TABLE)
    }

    override fun openWhitePageList() {
        openList(AdBlockManager.TYPE_WHITE_PAGE_TABLE)
    }

    private fun openList(type: Int) {
        supportFragmentManager.beginTransaction()
                .replace(R.id.container, AdBlockFragment(type), TAG_LIST)
                .addToBackStack("type:$type")
                .commit()
    }

    override fun onImport(adBlocks: List<AdBlock>) {
        val fragment = supportFragmentManager.findFragmentByTag(TAG_LIST)
        (fragment as? AdBlockFragment)?.addAll(adBlocks)
    }


    override fun setFragmentTitle(type: Int) {
        when (type) {
            AdBlockManager.TYPE_BLACK_TABLE -> setTitle(R.string.pref_ad_block_black)
            AdBlockManager.TYPE_WHITE_TABLE -> setTitle(R.string.pref_ad_block_white)
            AdBlockManager.TYPE_WHITE_PAGE_TABLE -> setTitle(R.string.pref_ad_block_white_page)
        }
    }

    override fun requestImport(uri: Uri) {
        supportFragmentManager.beginTransaction()
                .replace(R.id.container, AdBlockImportFragment(uri))
                .addToBackStack("")
                .commit()
    }

    override fun getExportFileName(type: Int) = when (type) {
        AdBlockManager.TYPE_BLACK_TABLE -> "black_list.txt"
        AdBlockManager.TYPE_WHITE_TABLE -> "white_list.txt"
        AdBlockManager.TYPE_WHITE_PAGE_TABLE -> "white_page_list.txt"
        else -> throw IllegalArgumentException()
    }

    companion object {
        const val ACTION_OPEN_BLACK = "jp.hazuki.yuzubrowser.adblock.AdBlockActivity.action.open.black"
        const val ACTION_OPEN_WHITE = "jp.hazuki.yuzubrowser.adblock.AdBlockActivity.action.open.white"
        const val ACTION_OPEN_WHITE_PAGE = "jp.hazuki.yuzubrowser.adblock.AdBlockActivity.action.open.whitepage"

        private const val TAG_LIST = "list"
    }
}
